bootstrap_salt/fab_tasks.py

```diff
@@ -97,35 +97,30 @@ def set_master_dns():
     return True
 
 
-def get_candidate_minions(stack_name):
-    cfn = get_connection(Cloudformation)
-    ec2 = get_connection(EC2)
-    instance_ids = cfn.get_stack_instance_ids(stack_name)
-    master_instance_id = ec2.get_master_instance(stack_name).id
-    instance_ids.remove(master_instance_id)
-    return instance_ids
-
-
 def install_minions():
     _validate_fabric_env()
     stack_name = get_stack_name()
     ec2 = get_connection(EC2)
     print "Waiting for SSH on all instances..."
     ec2.wait_for_ssh(stack_name)
-    candidates = get_candidate_minions(stack_name)
-    existing_minions = ec2.get_minions(stack_name)
-    to_install = list(set(candidates).difference(set(
-        [x.id for x in existing_minions])))
-    if not to_install:
-        return
-    public_ips = ec2.get_instance_public_ips(to_install)
-    sha = '6080a18e6c7c2d49335978fa69fa63645b45bc2a'
+
     master_inst = ec2.get_master_instance(stack_name)
     master_public_ip = master_inst.ip_address
     master_prv_ip = master_inst.private_ip_address
-    ec2.set_instance_tags(to_install, {'SaltMasterPrvIP': master_prv_ip})
-    for inst_ip in public_ips:
-        env.host_string = 'ubuntu@%s' % inst_ip
+
+    to_install = ec2.get_unconfigured_minions(stack_name, master_prv_ip)
+
+    if not to_install:
+        return
+
+    sha = '6080a18e6c7c2d49335978fa69fa63645b45bc2a'
+    for instance in to_install:
+
+        if env.user:
+            env.host_string = instance.ip_address
+        else:
+            env.host_string = "{0}@{1}".format(env.user, instance.ip_address)
```

Comment:
This was all sort of wrong - good spot. It has been pulled out

(env.user gets defaulted by fabric to be the current user *and* this would have been a potentially breaking change.)